### DISCUSSION TITLE
is there a way i can check the expiration for apple subscription purchase ?

### DISCUSSION THREAD

**Discussion by skakrecha** [imdhemy/laravel-in-app-purchases#75] [Q&A] [answered]
I am able to check expiration for google subscription 
like this 
`$subscriptionReceipt = OnlineSubscription::googlePlay()->id($active_package->store_id)->token($active_package->transaction_id)->get();
                if($subscriptionReceipt &&  ($subscriptionReceipt->getExpiryTime()->isPast() )){
              // Do Something
}`

is there a way i can do the same with apple subscription ?

like this 
`    $subscriptionReceipt = Subscription::appStore()->receiptData($payload)->verifyReceipt();
    return $subscriptionReceipt->getExpiryTime();`

**Comment by skakrecha** [comment]
@imdhemy hey can you help me with this. thanks

**Reply by imdhemy** [reply]
@skakrecha Noted... I Will answer this question and other questions after my work hours today! keep tuned!

**Reply by skakrecha** [reply]
ok thanks brother ! :)

**Comment by imdhemy** [comment, marked as answer]
@skakrecha 
In recent updates, I added a `toStd()` method to the `Subscription` facade. It makes the process cleaner and easier for both Google Play and App Store. You can use it as follows:

# Goole Play

```php
$subscription = Subscription::googlePlay()->id($itemId)->token($token)->toStd();
$isExpired = $subscription->getExpiryTime()->isPast();
```

# App Store
```php
$subscription = Subscription::googlePlay()->receiptData($payload)->toStd();
$isExpired = $subscription->getExpiryTime()->isPast();
```

It's good to mention that the `Time` object returned from the `getExpiryTime()` method can be converted to a `Carbon` object or a `Datetime` object as needed:

```php
$expiryTime = $subscription->getExpiryTime();
$carbon = $expiryTime->getCarbon();
$dateTime = $expiryTime->toDateTime();
```

**Reply by skakrecha** [reply]
hey @imdhemy thanks alot brother. you have made my life easier 👍

**Reply by imdhemy** [reply]
@skakrecha You are welcome! ^^
